javascript/i18n/phonenumbers/demo.js

```diff
@@ -46,75 +46,82 @@ function phoneNumberParser() {
-          output.append('INVALID_COUNTRY_CODE');
-          break;
-        case PNV.TOO_SHORT:
-          output.append('TOO_SHORT');
-          break;
-        case PNV.TOO_LONG:
-          output.append('TOO_LONG');
-          break;
-      }
-      // IS_POSSIBLE shouldn't happen, since we only call this if _not_
-      // possible.
-      output.append('\nNote: numbers that are not possible have type ' +
-          'UNKNOWN, an unknown region, and are considered invalid.');
-    } else {
-      var isNumberValid = phoneUtil.isValidNumber(number);
-      output.append('\nResult from isValidNumber(): ');
-      output.append(isNumberValid);
-      if (isNumberValid && regionCode && regionCode != 'ZZ') {
-        output.append('\nResult from isValidNumberForRegion(): ');
-        output.append(phoneUtil.isValidNumberForRegion(number, regionCode));
-      }
-      output.append('\nPhone Number region: ');
-      output.append(phoneUtil.getRegionCodeForNumber(number));
-      output.append('\nResult from getNumberType(): ');
-      var PNT = i18n.phonenumbers.PhoneNumberType;
-      switch (phoneUtil.getNumberType(number)) {
-        case PNT.FIXED_LINE:
-          output.append('FIXED_LINE');
-          break;
-        case PNT.MOBILE:
-          output.append('MOBILE');
-          break;
-        case PNT.FIXED_LINE_OR_MOBILE:
-          output.append('FIXED_LINE_OR_MOBILE');
-          break;
-        case PNT.TOLL_FREE:
-          output.append('TOLL_FREE');
-          break;
-        case PNT.PREMIUM_RATE:
-          output.append('PREMIUM_RATE');
-          break;
-        case PNT.SHARED_COST:
-          output.append('SHARED_COST');
-          break;
-        case PNT.VOIP:
-          output.append('VOIP');
-          break;
-        case PNT.PERSONAL_NUMBER:
-          output.append('PERSONAL_NUMBER');
-          break;
-        case PNT.PAGER:
-          output.append('PAGER');
-          break;
-        case PNT.UAN:
-          output.append('UAN');
-          break;
-        case PNT.UNKNOWN:
-          output.append('UNKNOWN');
-          break;
-      }
+    output.append('\nResult from isPossibleNumberWithReason(): ');
+    var PNV = i18n.phonenumbers.PhoneNumberUtil.ValidationResult;
+    switch (phoneUtil.isPossibleNumberWithReason(number)) {
+      case PNV.INVALID_COUNTRY_CODE:
+        output.append('INVALID_COUNTRY_CODE');
+        break;
+      case PNV.TOO_SHORT:
+        output.append('TOO_SHORT');
+        break;
+      case PNV.TOO_LONG:
+        output.append('TOO_LONG');
+        break;
+      case PNV.IS_POSSIBLE:
+        output.append('IS_POSSIBLE');
+        break;
+      case PNV.IS_POSSIBLE_LOCAL_ONLY:
+        output.append('IS_POSSIBLE_LOCAL_ONLY');
+        break;
+      case PNV.INVALID_LENGTH:
+        output.append('INVALID_LENGTH');
+        break;
+    }
+    output.append('\nNote: numbers that are not possible have type ' +
+        'UNKNOWN, an unknown region, and are considered invalid.');
```

Comment:
This output was here so you did NOT have to then call isValidNumber for the non-possible case and print the stuff below. It makes no sense to say this and then print it.

Just change to if (isPossibleResult.equals(IS_POSSIBLE)) {
  // same code as before
}

Reply:
True.. As I have reverted the change and having the check possible check, leaving as is.
Please review and suggest.